dist/rules/no-internal.js

```diff
@@ -72,13 +80,17 @@ module.exports = {
       return undefined;
     }
 
-    function isLocalFile(declaration) {
-      if (declaration) {
-        const fileName = getFileName(declaration.parent);
-        if (fileName && typeof fileName === "string" && !fileName.includes("node_modules"))
-          return true;
-      }
-      return false;
+    /** @param {string} fileName */
+    function isLocalFile(fileName) {
+      return fileName && typeof fileName === "string" && !fileName.includes("node_modules");
+    }
+
+    function isCheckedFile(declaration) {
+      if (!declaration)
+        return false;
+      const fileName = getFileName(declaration.parent);
+      const inCheckedPackage = checkedPackagePatterns.some(p => new RegExp("node_modules/"+escapeRegExp(p)).test(fileName));
```

Comment:
Yeah originally I was building these earlier but for some reason thought I was doing something dynamic and moved it in here. Will go back to building them at init.

But I have always assumed because of that expense that there is an engine-level cache, but have never tested it.

Reply:
done